### PR TITLE
Improve display of commands

### PR DESCRIPTION
**Pull request by sstok**
|Q            |A  |
|---          |---|
|Fixed Tickets|   |
|License      |MIT|
                   
Work in progress

**Issue:show**
Issue:show open
Issue:show closed

**pull-request:list**
schermafbeelding 2015-11-13 om 15 41 44

**pull-request:list:milestones**
schermafbeelding 2015-11-13 om 15 42 14

### CODE REVIEW

**Comment by sstok** [comment]
Assignee is missing in issue:show :scream_cat: (Friday the 13th :joy:  )

File: composer.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
         "ext-curl": "*",
         "guzzlehttp/guzzle": "~4.2.3",
         "guzzlehttp/progress-subscriber": "~1.1.0",
-        "symfony/console": "~2.7.3",
+        "symfony/console": "~2.8.0@dev",
```
Comment (sstok) [inline composer.json]: This is required as the SymfonyStyle changed in 2.8, before this text was prefixed with `// `

**Comment by sstok** [comment]
And status will be handy also, argg :angry:

File: src/Command/PullRequest/PullRequestMilestoneListCommand.php [hunk 2]
```diff
@@ -42,31 +41,19 @@ protected function configure()
...
+                $input->getOption('org'), $input->getOption('repo')
+            )
+        );
 
-    private function getRowBuilderCallback()
-    {
-        return function ($milestone) {
-            return [$milestone];
-        };
+        $styleHelper->listing($milestones);
```
Comment (cordoval) [inline src/Command/PullRequest/PullRequestMilestoneListCommand.php]: listing -> list

**Comment by cordoval** [comment]
btw nice screenshots, looks like you moved to apple, way to go :+1:

File: src/Command/PullRequest/PullRequestMilestoneListCommand.php (hunk 2 above)
Comment (sstok) [inline src/Command/PullRequest/PullRequestMilestoneListCommand.php]: This simply an adapter/proxy for the SymfonyStyle class ;)

**Comment by sstok** [comment]
Now I only need to figure out what was is missing in this pr 😄

**Comment by sstok** [comment]
@cordoval ping

**Comment by cordoval** [comment]
after i come back i will test on hospice project, then will merge, i cannot merge because i need to test in real life scenario first.

File: src/Command/Issue/IssueShowCommand.php [hunk 3]
```diff
@@ -52,7 +54,18 @@ protected function configure()
     protected function execute(InputInterface $input, OutputInterface $output)
     {
         if (null === $issueNumber = $input->getArgument('issue')) {
-            $issueNumber = $this->getHelper('git')->getIssueNumber();
```
Comment (cordoval) [inline src/Command/Issue/IssueShowCommand.php]: i am wondering why this does not work well... I am doing some reviews here https://github.com/sstok/gush/pull/8

File: src/Helper/GitHelper.php [hunk 4]
```diff
@@ -278,18 +278,6 @@ public function listFiles($options = [])
         return StringUtil::splitLines($process->getOutput());
     }
 
-    public function getIssueNumber()
```
Comment (cordoval) [inline src/Helper/GitHelper.php]: i see now, let's move it back here

File: src/Command/Issue/IssueShowCommand.php [hunk 5]
```diff
@@ -52,7 +54,18 @@ protected function configure()
...
+            $segments = explode('-', $this->getHelper('git')->getActiveBranchName(), 2);
+
+            if (!isset($segments[1])) {
+                throw new UserException(
```
Comment (cordoval) [inline src/Command/Issue/IssueShowCommand.php]: the exception here is fine

File: tests/Command/Issue/IssueShowCommandTest.php [hunk 6]
```diff
@@ -17,55 +17,86 @@
...
+            'Gush\Exception\UserException',
+            'Unable to extract issue-number from the current branch.'
+        );
+
+        $tester->execute();
+    }
+
+    private function getLocalGitHelper($branch = 'test_branch')
     {
-        $helper = $this->getGitHelper(true);
```
Comment (cordoval) [inline tests/Command/Issue/IssueShowCommandTest.php]: this one is going to fail with the revisions i made, not sure yet how to tweak it

**Comment by cordoval** [comment]
introducing this changes on https://github.com/sstok/gush/pull/8

**Pull request closed without merging**